### PR TITLE
feat: router service

### PR DESCRIPTION
**Pull request by brucexc**

### CODE REVIEW

File: internal/hub/model/node_activity.go [hunk 1]
```diff
@@ -224,3 +226,98 @@ func BuildPath(path string, query any, nodes []Cache) (map[common.Address]string
...
+	Address common.Address
+	Data    []byte
+	// Valid indicates whether the data is non-null.
+	Valid          bool
+	Err            error
+	Request        int
+	InvalidRequest int
+}
+
+type ErrResponse struct {
```
Comment (HenryQW) [inline internal/hub/model/node_activity.go:240]: should provide a generic `ErrorResponse` struct

**Review by HenryQW — CHANGES_REQUESTED** [review]

**Review by HenryQW — CHANGES_REQUESTED** [review]
Also comments for functions are missing, especially long ones like `distribute` and `sendResult`

File: internal/router/router.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	}
+
+	if err := json.Unmarshal(data, target); err == nil {
+		return true
+	}
+
+	return false
+}
+
+func NewSimpleRouter(httpClient httpx.Client) (*SimpleRouter, error) {
```
[the remark this answers is not in the document]
Comment (brucexc) [reply internal/router/router.go:191]: given that the current implementation does not  generate an error, it's indeed redundant to include an error in the return values.

File: internal/router/router.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+			firstResult <- *result
+
+			*resultSent = true
+
+			return
+		}
+
+		// If all the results have been received
+		if len(*results) == nodeMapLen {
+			for _, res := range *results {
```
[the remark this answers is not in the document]
Comment (brucexc) [reply internal/router/router.go:158]: all error scenarios are indeed taken into consideration, and the first result will be returned in such cases

File: internal/hub/model/node_activity.go (hunk 1 above)
Comment (brucexc) [reply internal/hub/model/node_activity.go]: this depends on the error responses returned by the node 
https://github.com/RSS3-Network/Node/blob/develop/common/http/response/error.go#L20

File: internal/router/router.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+				return
+			}
+
+			firstResult <- *(*results)[0]
+		}
+	}
+}
+
+func validateData(data []byte, target any) bool {
+	var errRes model.ErrResponse
```
[the remark this answers is not in the document]
Comment (brucexc) [reply internal/router/router.go:179]: NotFoundResponse does not belong to the correct return path of the node and has been removed

**Review by HenryQW — APPROVED** [review]

**Pull request merged**